src/main/scala/mesosphere/marathon/core/deployment/impl/TaskReplaceActor.scala

```diff
@@ -111,12 +111,13 @@ class TaskReplaceActor(
   private def initialized: Receive = readinessBehavior orElse replaceBehavior
 
   def replaceBehavior: Receive = {
-    // New instance failed to start, restart it
-    case InstanceChanged(id, runSpecVersion, `pathId`, condition, Instance(instanceId, Some(agentInfo), state, tasksMap, runSpec, reservation)) if newInstanceIds(id) && considerTerminal(condition) =>
-      logger.warn(s"New instance $id is terminal on agent ${agentInfo.agentId} during app $pathId restart: $condition reservation: $reservation")
-      instanceTerminated(id)
-      instancesStarted -= 1
-      launchInstances().pipeTo(self)
+    // New instance task failed: instance goal is still Running so nothing to do here, the task will be rescheduled automatically
+    case InstanceChanged(id, runSpecVersion, `pathId`, condition, Instance(instanceId, Some(agentInfo), state, tasksMap, runSpec, reservation)) if newInstanceIds(id) && considerTerminal(condition) && state.goal == Goal.Running =>
+      logger.warn(s"New $id is terminal ($condition) on agent ${agentInfo.agentId} during app $pathId restart: $condition reservation: $reservation. Waiting for the task to restart...")
```

Comment:
Bug nr. 2: If a new instance task would fail, we would create a new one (`launchInstances()` call on the left side). This is not necessary anymore since the instance is already new and the task will be rescheduled by the `TaskLauncherActor`